src/storage-blob-preview/azext_storage_blob_preview/operations/blob.py

```diff
@@ -195,64 +191,86 @@ def _download_blob(blob_service, container, destination_folder, normalized_blob_
                            'command instead to download individual blobs.'.format(normalized_blob_name))
         blobs_to_download[normalized_blob_name] = blob_name
 
+    results = []
     if dryrun:
+        # download_blobs = _blob_precondition_check(source_blobs, if_modified_since=if_modified_since,
+        #                                           if_unmodified_since=if_unmodified_since)
         logger.warning('download action: from %s to %s', source, destination)
         logger.warning('    pattern %s', pattern)
-        logger.warning('  container %s', source_container_name)
+        logger.warning('  container %s', container_name)
         logger.warning('      total %d', len(source_blobs))
         logger.warning(' operations')
         for b in source_blobs:
             logger.warning('  - %s', b)
-        return []
 
-    # Tell progress reporter to reuse the same hook
-    if progress_callback:
-        progress_callback.reuse = True
+    else:
+        @check_precondition_success
+        def _download_blob(*args, **kwargs):
+            blob = download_blob(*args, **kwargs)
+            return blob.name
 
-    results = []
-    for index, blob_normed in enumerate(blobs_to_download):
-        # add blob name and number to progress message
+        # Tell progress reporter to reuse the same hook
         if progress_callback:
-            progress_callback.message = '{}/{}: "{}"'.format(
-                index + 1, len(blobs_to_download), blobs_to_download[blob_normed])
-        results.append(_download_blob(
-            client, source_container_name, destination, blob_normed, blobs_to_download[blob_normed]))
+            progress_callback.reuse = True
 
-    # end progress hook
-    if progress_callback:
-        progress_callback.hook.end()
+        for index, blob_normed in enumerate(blobs_to_download):
+            # add blob name and number to progress message
+            if progress_callback:
+                progress_callback.message = '{}/{}: "{}"'.format(
+                    index + 1, len(blobs_to_download), blobs_to_download[blob_normed])
+            blob_client = client.get_blob_client(container=container_name,
+                                                 blob=blobs_to_download[blob_normed])
+            destination_path = os.path.join(destination, os.path.normpath(blob_normed))
+            destination_folder = os.path.dirname(destination_path)
+            # Failed when there is same name for file and folder
+            if os.path.isfile(destination_folder) and os.path.exists(destination_folder):
+                raise CLIError("There is a file having the same name with directory to create. Please rename the file "
+                               "and retry the command.")
```

Comment:
Do we need to print out the duplicate file name here?
By the way, I have a little suggestion: maybe we can consider adding options to support users to choose to overwrite or skip the path with the same name. In this way, the whole download process will not be interrupted by one of the duplicate name of folder or file.

Reply:
print file name here. But I don't want to skip them because it may change import files existed in the folder for customer.